### PR TITLE
upgrade to react-native v0.35.0

### PR DESCRIPTION
**Pull request by justinwoo**
Let's upgrade React-Native to v0.35.0 before we can get other things rolling.

Please make sure this "makes sense", mostly just looking at this on local iOS/Android simulators and checking the counter and stuff.

### CODE REVIEW

**Comment by justinwoo** [comment]
Ignore branch name; this is now 0.35.0.

As before, please check this works!

**Comment by tino-junge** [comment]
Build succeeds for iOS but app won't start. In XCode I receive the following error log:

```
2016-10-11 22:22:19.178 [info][tid:main][RCTBatchedBridge.m:72] Initializing <RCTBatchedBridge: 0x7faad400d360> (parent: <RCTBridge: 0x7faad2420ec0>, executor: RCTJSCExecutor)
2016-10-11 22:22:19.402 [error][tid:com.facebook.react.JavaScript] JSON Parse error: Unexpected identifier "RCTWebSocketModule"
2016-10-11 22:22:19.403 [info][tid:com.facebook.react.JavaScript] 'Failed to print error: ', 'JSON Parse error: Unexpected identifier "RCTSourceCode"'
2016-10-11 22:22:19.419 [error][tid:com.facebook.react.JavaScript] Requiring module "236", which threw an exception.
2016-10-11 22:22:19.419 [info][tid:com.facebook.react.JavaScript] 'Failed to print error: ', 'Requiring module "242", which threw an exception.'
2016-10-11 22:22:19.420 [info][tid:main][RCTRootView.m:239] Running application PepperoniAppTemplate ({
    initialProps =     {
    };
    rootTag = 1;
})
2016-10-11 22:22:19.425 [error][tid:com.facebook.react.JavaScript] Module AppRegistry is not a registered callable module.
2016-10-11 22:22:19.425 [error][tid:com.facebook.react.JavaScript] Requiring module "242", which threw an exception.
2016-10-11 22:22:19.425 [info][tid:com.facebook.react.JavaScript] 'Failed to print error: ', 'Requiring module "242", which threw an exception.'
2016-10-11 22:22:19.425 [info][tid:com.facebook.react.JavaScript] 'Failed to print error: ', 'JSON Parse error: Unexpected identifier "RCTExceptionsManager"'

```

**Comment by justinwoo** [comment]
"works on my machine": I'm able to start the app and click around and everything. Have you tried wiping your node modules and whatnot?

**Comment by tino-junge** [comment]
Finally got it working. Removing node_nodules wasn't enough though. had to reset npm cache as well. The following commands solved it:
```
rm -rf node_modules && npm i
npm start -- --reset-cache
```

Good to go from my site! 👍

**Comment by hhimanshu** [comment]
Can we have this merged please? Currently with xCode 8, this does not work

File: ios/PepperoniAppTemplate.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -22,9 +22,7 @@
 		13B07FC11A68108700A75B9A /* main.m in Sources */ = {isa = PBXBuildFile; fileRef = 13B07FB71A68108700A75B9A /* main.m */; };
 		140ED2AC1D01E1AD002B40FF /* libReact.a in Frameworks */ = {isa = PBXBuildFile; fileRef = 146834041AC3E56700842450 /* libReact.a */; };
 		146834051AC3E58100842450 /* libReact.a in Frameworks */ = {isa = PBXBuildFile; fileRef = 146834041AC3E56700842450 /* libReact.a */; };
-		1916AD410B0B45C9BEDB553E /* libA0RNLock.a in Frameworks */ = {isa = PBXBuildFile; fileRef = CBED167597174594A47B20BE /* libA0RNLock.a */; };
```
Comment (krivachy) [inline ios/PepperoniAppTemplate.xcodeproj/project.pbxproj]: It seems that you are removing all Auth0 dependencies, this kind-of breaks Auth0 :)

Here's what I get when configuring it (i.e. it's confirmed to be broken):

```
69Possible Unhandled Promise Rejection (id: 0):
Cannot read property 'registerColorForKey' of undefined
TypeError: Cannot read property 'registerColorForKey' of undefined
    at http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false:97042:11
    at Array.forEach (native)
    at Auth0Lock.customizeTheme (http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false:97040:22)
    at Object.showLogin (http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false:96935:6)
    at Constructor.componentWillReceiveProps (http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false:96418:7)
    at http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false:27229:13
    at measureLifeCyclePerf (http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false:26695:8)
    at ReactCompositeComponentWrapper.updateComponent (http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false:27228:1)
    at ReactCompositeComponentWrapper.receiveComponent (http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false:27165:6)
    at Object.receiveComponent (http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false:25340:18)
    at ReactCompositeComponentWrapper._updateRenderedComponent (http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false:27372:17)
    at ReactCompositeComponentWrapper._performComponentUpdate (http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false:27342:6)
    at ReactCompositeComponentWrapper.updateComponent (http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false:27263:6)
    at ReactCompositeComponentWrapper.performUpdateIfNecessary (http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false:27179:6)
    at Object.performUpdateIfNecessary (http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false:25372:18)
    at runBatchedUpdates (http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false:24983:17)
4debugger-ui:1 Error in event handler for (unknown): TypeError: Cannot read property 'shortcutLock' of undefined
    at chrome-extension://oknpjjbmpnndlpmnhmekjpocelpnlfdi/js/content.js:32:32
```

**Comment by krivachy** [comment]
@hhimanshu we're working on merging this, but the framework requires thorough testing on iOS and Android which usually takes up to an hour each time as we start from a fresh clone to simulate a new developer picking up the tool. In the meantime you can check out or merge this branch wherever you want if you are willing to live a bit on the edge :) Issues I found are commented here, so you'll know what you'll be getting.

File: .gitignore [hunk 2]
```diff
@@ -24,23 +24,16 @@ project.xcworkspace
...
-*.iml
 
 # node.js
 #
 node_modules/
-coverage/
 npm-debug.log
 
-# Secrets
-env.js
```
Comment (chadrien) [inline .gitignore]: I think it should. Also this PR contains a `env.js` file that should be removed as well 🙂

**Comment by krivachy** [comment]
@justinwoo Please fix the commit message, and then it's 👍

**Pull request merged**